adminpages/memberslist-csv.php

```diff
@@ -35,10 +35,22 @@
 	global $wpdb;
 
 	//get users (search input field)
-	if(isset($_REQUEST['s']))
-		$s = sanitize_text_field($_REQUEST['s']);
-	else
-		$s = "";
+	$search_key = false;
+	if( isset( $_REQUEST['s'] ) ) {
+		$s = sanitize_text_field( trim( $_REQUEST['s'] ) );
+	} else {
+		$s = '';
+	}
+
+	// If there's a colon in the search, let's split it out.
+	if( ! empty( $s ) && strpos( $s, ':' ) !== false ) {
+		$parts = explode( ':', $s );
+		$search_key = $parts[0];
+		$s = $parts[1];
```

Comment:
Should we use `array_shift()` here so that subsequent `:` will get ignored but still used in the search string itself? Does it matter?

```suggestion
		$search_key = array_shift( $parts );
		$s = implode( ':', $parts );
```